packages/pigeon/lib/generator_tools.dart

```diff
@@ -249,3 +249,42 @@ Map<String, Object> mergeMaps(
   }
   return result;
 }
+
+/// A class name that is enumerated.
+class EnumeratedClass {
+  /// Constructor.
+  EnumeratedClass(this.name, this.enumeration);
+
+  /// The name of the class.
+  final String name;
+
+  /// The enumeration of the class.
+  final int enumeration;
+}
+
+/// Custom codecs' custom types are enumerated from 255 down to this number to
```

Comment:
Iterating down rather than up seems odd now that you're no longer in a contested ID space.

Reply:
done